service-app/internal/httpclient/request.go

```diff
@@ -13,6 +14,8 @@ import (
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 )
 
+var NotFoundError = errors.New("received 404 response from")
```

Comment:
```suggestion
var ErrNotFound = errors.New("received 404 response from")
```

is the usual naming for instances of an error